lib/srv/db/watcher.go

```diff
@@ -53,12 +53,17 @@ func (s *Server) startReconciler(ctx context.Context) error {
 		for {
 			select {
 			case <-s.reconcileCh:
-				if err := reconciler.Reconcile(ctx); err != nil {
-					s.log.ErrorContext(ctx, "Failed to reconcile.", "error", err)
-				}
-				if s.cfg.OnReconcile != nil {
-					s.cfg.OnReconcile(s.getProxiedDatabases())
-				}
+				func() {
```

Comment:
Is the separate function call just to be able to do a "defer"? That's a little confusing, can we just do mutex unlock after without defer?